flexmeasures/cli/monitor.py

```diff
@@ -257,10 +243,21 @@ def monitor_last_seen(
         users = [user for user in users if user.account.has_role(account_role)]
     if user_role is not None:
         users = [user for user in users if user.has_role(user_role)]
-
+    # filter out users who we already included in this check's last run
+    txt_about_ignored_users = ""
+    if not include_all_users_each_run and latest_run:
+        original_length = len(users)
+        users = [
+            user
+            for user in users
+            if user.last_seen_at.replace(tzinfo=timezone.utc) + last_seen_delta
+            > latest_run.datetime
```

Comment:
I tested it and the code works without a previous run. I believe datetimes can be compared to null.

Reply:
I meant it differently. As far as I understand, this code assumes that if a timespan has already passed in which a notification *should have* previously been sent, it will skip sending a new notification. However, there is no explicit guarantee that a notification had actually been sent in the timespan in which it should have been sent.